presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnChecksum.java

```diff
@@ -63,19 +78,29 @@ public boolean equals(Object obj)
         }
         ArrayColumnChecksum o = (ArrayColumnChecksum) obj;
         return Objects.equals(checksum, o.checksum) &&
+                Objects.equals(floatingPointChecksum, o.floatingPointChecksum) &&
                 Objects.equals(cardinalityChecksum, o.cardinalityChecksum) &&
                 Objects.equals(cardinalitySum, o.cardinalitySum);
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hash(checksum, cardinalityChecksum, cardinalitySum);
+        return Objects.hash(checksum, floatingPointChecksum, cardinalityChecksum, cardinalitySum);
     }
 
     @Override
     public String toString()
     {
-        return format("checksum: %s, cardinality_checksum: %s, cardinality_sum: %s", checksum, cardinalityChecksum, cardinalitySum);
+        if (!floatingPointChecksum.isPresent()) {
+            return format("checksum: %s, cardinality_checksum: %s, cardinality_sum: %s", checksum, cardinalityChecksum, cardinalitySum);
+        }
+        else {
+            return format(
+                    "%s, cardinality_checksum: %s, cardinality_sum: %s",
+                    floatingPointChecksum.get().toString(),
```

Comment:
fyi, you don't need to call toString() here. The format function will do it if you pass in the object.